Gemfile

```diff
@@ -14,63 +14,32 @@ group :documentation do
   gem 'relish', '~> 0.7.1'
 end
 
-platforms :jruby do
-  gem "jruby-openssl"
-end
+gem 'rake', '> 12'
 
-case RAILS_VERSION
-when /main/
-  MAJOR = 6
-  MINOR = 0
-when /5-2-stable/
-  MAJOR = 5
-  MINOR = 2
-when /stable/
-  MAJOR = 6
-  MINOR = 0
-when nil, false, ""
-  MAJOR = 6
-  MINOR = 0
-else
-  match = /(\d+)(\.|-)(\d+)/.match(RAILS_VERSION)
-  MAJOR, MINOR = match.captures.map(&:to_i).compact
+if RUBY_VERSION.to_f >= 2.3
+  gem 'rubocop', '~> 0.80.1'
 end
 
+gem 'capybara'
+
+MAJOR =
+  case RAILS_VERSION
+  when /5-2-stable/
+    5
+  when /main/, /stable/, nil, false, ''
+    6
+  else
+    /(\d+)[\.|-]\d+/.match(RAILS_VERSION).captures.first.to_i
+  end
+
 if MAJOR >= 6
-  gem 'selenium-webdriver', '~> 3.5', require: false
+  # sqlite3 is an optional, unspecified, dependency and Rails 6.0 only supports `~> 1.4`
   gem 'sqlite3', '~> 1.4', platforms: [:ruby]
```

Comment:
`example_app_generator/generate_app.rb` ends up with
```ruby
gem 'sqlite3'
```
in example_app's Gemfile. Why there's no version mismatch, and how the proper version is resolved is beyond my understanding.
I used a different approach for 3-9-maintenance branch, moved this selection to `Gemfile-rails-dependencies`, and added a statement to strip example_app's `Gemfile` from `gem 'sqlite3'`. https://github.com/rspec/rspec-rails/pull/2386/files#diff-8c3955be04b733dbd31e54e50844fb84